File: java_logic/calculadora.java
import java.util.Scanner; // Importa a classe para leitura de dados [ cite: 11]
import java.util.Locale; // Importa para garantir o uso do ponto decima em vez de vírgula

public class calculadora {   
    public static void main(String[] args) {
        // Configura o ponto como separador decimal (Padrão Americano/Programação)
        Locale.setDefault(Locale.US);

        // Passo 1: Instanciar o objeto Scanner para ler a entrada do teclado [cite: 11]
        Scanner sc = new Scanner(System.in);

        // Passo 2: Entrada de Dados e Declaração de Variáveis [cite: 6]
        System.out.println("---Calculadora de Negócios (Java)---");

        System.out.print("Digite o primeiro número:");
        double numero1 = sc.nextDouble();

        System.out.print("Digite o segundo numero:");
        double numero2 = sc.nextDouble();

        // Passo 3: Processamento (Cálculos Aritméticos) [cite: 7]
        double soma = numero1 + numero2;
        double subtracao = numero1 - numero2;
        double multiplicacao = numero1 * numero2;
        double divisao = numero1 / numero2;

        // Passo 4: Saída de Resultados Formatados
        System.out.printf("Soma:%.2f%n", soma);
        System.out.printf("Subtração:%.2f%n", subtracao);
        System.out.printf("Multiplicação:%.2f%n", multiplicacao);
        System.out.printf("Divisão:%.2f%n", divisao);

        //Passo 5: Fechar o Scanner (Boa prática de gerenciamento de memória)
        sc.close();
     }
}     

File: cpp_logic/calculadora.cpp
# include <iostream>
# include <iomanip>
using namespace std;

int main() {
    // Passo 1: Declaração de Variáveis ( Tipagem Estática)
    int numero1 = 0, numero2 = 0, soma, subtracao, multiplicacao;
    float divisao;

    // Passo 2: Entrada de Dados
    cout << "Digite o primeiro numero:"; cin >> numero1;
    cout << "Digite o segundo numero:"; cin >> numero2;

    // Passo 3 Processamento (Calculos Aritmeticos)
    soma = numero1 + numero2;
    subtracao = numero1 - numero2;
    multiplicacao = numero1 * numero2;
    divisao = numero1  / numero2;

    // Passo 4: Saida de Resultados Formatados
    cout <<fixed << setprecision(2);
    cout <<"Soma: "<< soma<< endl;
    cout <<"Subtracao: "<<subtracao <<endl;
    cout <<"Multiplicacao: "<<multiplicacao << endl;
    cout <<"Divisao: "<<divisao << endl;
    

    return 0;
  }